extensions/sql-database-projects/src/controllers/databaseProjectTreeViewProvider.ts

```diff
@@ -0,0 +1,132 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import * as nls from 'vscode-nls';
+
+import { SqlDatabaseProjectItem } from './databaseProjectTreeItem';
+import { Deferred } from '../common/promise';
+
+
+const localize = nls.loadMessageBundle();
+
+export class SqlDatabaseProjectTreeViewProvider implements vscode.TreeDataProvider<SqlDatabaseProjectItem> {
+	private _onDidChangeTreeData: vscode.EventEmitter<SqlDatabaseProjectItem | undefined> = new vscode.EventEmitter<SqlDatabaseProjectItem | undefined>();
+	readonly onDidChangeTreeData: vscode.Event<SqlDatabaseProjectItem | undefined> = this._onDidChangeTreeData.event;
+
+	private _initializeDeferred: Deferred<void> = new Deferred<void>();
+
+	private roots: SqlDatabaseProjectItem[] = [];
+
+	constructor() {
+		this._initializeDeferred = new Deferred<void>();
+		this.initialize();
+	}
+
+	async initialize() {
+		this.roots = [new SqlDatabaseProjectItem(localize('noProjectOpenMessage', "No open database project"), false)];
+
+
+		this._initializeDeferred.resolve();
+	}
+
+	getTreeItem(element: SqlDatabaseProjectItem): vscode.TreeItem {
```

Comment:
nit: But it's generally nice to specifically include the visibility modifier for class functions

Reply:
Of course.  I'm surprised the linter doesn't complain about this.  Worth a modification?